ooniprobe/Test/Test/WebConnectivity.mm

```diff
@@ -61,8 +69,12 @@ -(void)setUrls:(NSArray*)inputs{
 }
 
 -(void)setDefaultMaxRuntime {
-    NSNumber *max_runtime = [[NSUserDefaults standardUserDefaults] objectForKey:@"max_runtime"];
-    self.settings.options.max_runtime = max_runtime;
+    if ([[[NSUserDefaults standardUserDefaults] objectForKey:@"max_runtime_enabled"] boolValue]){
+        NSNumber *max_runtime = [[NSUserDefaults standardUserDefaults] objectForKey:@"max_runtime"];
+        self.settings.options.max_runtime = max_runtime;
+    }
+    else
+        self.settings.options.max_runtime = MAX_RUNTIME_DISABLED;
```

Comment:
So, reading ObjC code is easier. Why are we caching the `max_runtime` from settings to an attribute of the WebConnectivity class? What is the purpose?

Reply:
because a test has a setting class assigned